src/test/java/seedu/address/logic/commands/DeleteCommandTest.java

```diff
@@ -134,7 +146,57 @@ public void execute_invalidIndexFilteredList_throwsCommandException() {
 
         DeleteCommand deleteCommand = new DeleteCommand(outOfBoundIndex);
 
-        assertCommandFailure(deleteCommand, model, Messages.MESSAGE_INVALID_PERSON_DISPLAY_INDEX);
+        assertCommandFailure(deleteCommand, model, Messages.MESSAGE_INVALID_INDEX);
+    }
+
+    @Test
+    public void execute_notValidContext_throwsCommandException() {
+        DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST);
+        Model model = new ModelManager();
+
+        assertThrows(CommandException.class, () -> deleteCommand.execute(model));
+    }
+
+    @Test
+    public void execute_listActivityContext_deleteSuccess() throws CommandException {
+        ModelManager expectedModel = new ModelManager(
+                model2.getAddressBook(), new UserPrefs(), new InternalState(), model2.getActivityBook());
+        model2.setContext(listActivitiesContext);
+        DeleteCommand deleteCommand = new DeleteCommand(INDEX_FIRST);
+
+        expectedModel.setContext(listActivitiesContext);
+        Activity activityToDelete = model2.getFilteredActivityList().get(INDEX_FIRST.getZeroBased());
+        String expectedMessage = String.format(DeleteCommand.MESSAGE_DELETION_SUCCESS, activityToDelete);
+        expectedModel.deleteActivity(activityToDelete);
+
+        assertCommandSuccess(deleteCommand, model2, expectedMessage, expectedModel, listActivitiesContext);
+    }
+
+    @Test
+    public void execute_viewActivityContext_softDeleteExpenseSuccess() throws CommandException {
```

Comment:
You can consider adding one more test for when a user attempts to soft-delete an invalid expense index (this command should fail).